### PR TITLE
Simplify the logic of adding/moving article to reading list

### PR DESCRIPTION
**Pull request by cooltey**
This PR simplified the logic of adding/moving an article to the list, which now can use either `ReadingListBehaviorsUtil.addToDefaultList` or `ReadingListBehaviorsUtil.moveToList` with an optional dialog dismiss listener.

This PR also fixed the issue that some views have incorrect invoke sources implemented.

### CODE REVIEW

**Comment by cooltey** [comment]
I added `High Priority` label so this behavior can be used in #4334 after merging into the main branch.

**Comment by sharvaniharan** [comment]
Lgth... will let @dbrant take a look and merge!

File: app/src/main/java/org/wikipedia/page/linkpreview/LinkPreviewDialog.kt
Comment (dbrant) [inline app/src/main/java/org/wikipedia/page/linkpreview/LinkPreviewDialog.kt:225]: Should we call `dismiss()` here, so that the LinkPreviewDialog is dismissed and the snackbar can be seen?

**Review by dbrant — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/wikipedia/page/linkpreview/LinkPreviewDialog.kt
Comment (cooltey) [reply app/src/main/java/org/wikipedia/page/linkpreview/LinkPreviewDialog.kt:225]: Done!

**Comment by dbrant** [comment]
looks good, but will merge after Beta release, just to let this change breathe until next release.

**Pull request merged**
